src/python/zquantum/core/estimator_test.py

```diff
@@ -14,7 +43,7 @@ def setUp(self):
         self.operator = QubitOperator("Z0")
         self.circuit = Circuit(Program(X(0)))
         self.backend = MockQuantumBackend()
-        self.n_samples = None
+        self.n_samples = 10
```

Comment:
I think we should have also a separate test for testing the case with `self.n_samples=None`?

Reply:
Added a test in 4bcf9ed5c4e6c77779b66b2964a067e80643f8c0.

I wonder if we should remove `n_samples` as an attribute of backend so that we don't have two separate places to specify the same thing. (Although I don't think we need to resolve that in this PR.)